diff -Nur openam-1.13.5/main.cxx openam/main.cxx
--- openam-1.13.5/main.cxx	2004-01-31 08:38:21.000000000 +0100
+++ openam/main.cxx	2005-02-03 02:21:42.000000000 +0100
@@ -27,6 +27,26 @@
  * Contributor(s): ______________________________________.
  *
  * $Log$
+ * Revision 1.93  2005/02/03 01:21:42  csoutheren
+ * Added called party to call log and command parameters
+ *
+ * Revision 1.92  2004/05/26 04:01:57  csoutheren
+ * Changed to work completely with new plugin codecs
+ *
+ * Revision 1.91  2004/05/10 13:07:19  rjongbloed
+ * Changed G.726 and MS-ADPCM to plug in codecs.
+ *
+ * Revision 1.90  2004/05/04 12:21:16  rjongbloed
+ * Converted LPC-10 codec to plug in.
+ *
+ * Revision 1.89  2004/05/03 13:25:46  rjongbloed
+ * Converted everything to be codec plug in freindly
+ * Removed GSM and G.729 as now plug ins are "the way"!
+ *
+ * Revision 1.88  2004/04/06 11:27:49  rjongbloed
+ * Changes to support native C++ Run Time Type Information
+ * Changes for codec plug ins
+ *
  * Revision 1.87  2004/01/31 07:38:21  rjongbloed
  * Changed HAS_IXJ usage to if defined not if non-zero
  *
@@ -326,10 +346,7 @@
 #include <ptlib/pipechan.h>
 
 #include "version.h"
-#include "lpc10codec.h"
-#include "speexcodec.h"
 
-#include "mscodecs.h"
 #include "opalvxml.h"
 #include "main.h"
 
@@ -378,10 +395,11 @@
                     const PString & user,
                     unsigned len,
                     const PString & codec,
-                    const PString & product)
+                    const PString & product,
+                    const PString & to)
 {
   PString addr = from;
-  LogMessage(addr & "\"" + user + "\"" & PString(PString::Unsigned, len) & codec & "\"" + product + "\"" & "\"" + fn + "\"");
+  LogMessage(addr & "\"" + user + "\"" & PString(PString::Unsigned, len) & codec & "\"" + product + "\"" & "\"" + fn + "\"" & "\"" + to + "\"");
 }
 
 
@@ -802,38 +820,30 @@
     return FALSE;
   }
 
-  if (!g7231Ogm.IsEmpty())
-    SetCapability(0, 0, new G7231_File_Capability);
+  AddAllCapabilities(0, 0, "*");
 
-  if (!gsmOgm.IsEmpty())
-    SetCapability(0, 0, new H323_GSM0610Capability);
+  PString removeString;
+  if (gsmOgm.IsEmpty())
+    removeString = removeString & OpalGSM0610 & "MS-GSM";
+
+  if (speexOgm.IsEmpty())
+    removeString = removeString & "Speex";
+ 
+  if (g711Ogm.IsEmpty())
+    removeString = removeString & "711";
 
-  if (!gsmOgm.IsEmpty())
-    SetCapability(0, 0, new MicrosoftGSMAudioCapability);
+  if (lpc10Ogm.IsEmpty())
+    removeString = removeString & "LPC-10";
 
-  if (!g711Ogm.IsEmpty())
-    SetCapability(0, 0, new H323_G711Capability(H323_G711Capability::muLaw, H323_G711Capability::At64k));
+  if (ilbcOgm.IsEmpty())
+    removeString = removeString & "iLBC";
 
-  if (!g711Ogm.IsEmpty())
-    SetCapability(0, 0, new H323_G711Capability(H323_G711Capability::ALaw, H323_G711Capability::At64k));
-
-  if (!lpc10Ogm.IsEmpty())
-    SetCapability(0, 0, new H323_LPC10Capability(*this));
+  if (!g7231Ogm.IsEmpty())
+    SetCapability(0, 0, new G7231_File_Capability);
 
-  if (!speexOgm.IsEmpty()) {
-    SetCapability(0, 0, new SpeexNarrow2AudioCapability());
-    SetCapability(0, 0, new SpeexNarrow3AudioCapability());
-    SetCapability(0, 0, new SpeexNarrow4AudioCapability());
-    SetCapability(0, 0, new SpeexNarrow5AudioCapability());
-    SetCapability(0, 0, new SpeexNarrow6AudioCapability());
-  }
-  
-  if (!ilbcOgm.IsEmpty()) {
-    SetCapability(0, 0, new H323_iLBC_Capability(*this,H323_iLBC_Capability::e_13k3));
-    SetCapability(0, 0, new H323_iLBC_Capability(*this,H323_iLBC_Capability::e_15k2));
-  }
-  
   capabilities.Remove(args.GetOptionString('D').Lines());
+  if (!removeString.IsEmpty())
+    capabilities.Remove(removeString.Tokenise(' '));
   capabilities.Reorder(args.GetOptionString('P').Lines());
 
   cout << "Codecs (in preference order):\n" << setprecision(2) << capabilities << endl;
@@ -1058,6 +1068,20 @@
 
 }
 
+BOOL MyH323Connection::OnReceivedSignalSetup(const H323SignalPDU & setupPDU)
+{
+  if (!H323Connection::OnReceivedSignalSetup(setupPDU))
+    return FALSE;
+
+  // extract the called number
+  const H225_Setup_UUIE & setup = setupPDU.m_h323_uu_pdu.m_h323_message_body;
+  if (setup.HasOptionalField(H225_Setup_UUIE::e_destinationAddress) && setup.m_destinationAddress.GetSize() > 0) {
+    calledParty = H323GetAliasAddressString(setup.m_destinationAddress[0]);
+  }
+
+  return TRUE;
+}
+
 MyH323Connection::~MyH323Connection()
 {
   cout << "Closing connection" << endl;
@@ -1069,7 +1093,7 @@
   PString codecStr = receiveCodecName + "/" + transmitCodecName;
   unsigned duration = (unsigned)((interval.GetMilliSeconds()+999)/1000);
 
-  LogCall(recordFn, addr, GetRemotePartyName(), duration, codecStr, product);
+  LogCall(recordFn, addr, GetRemotePartyName(), duration, codecStr, product, calledParty);
 
   if ((recordFile!= NULL) && (recordFile->WasRecordStarted()) && !ep.GetRunCmd().IsEmpty()) {
     PString cmdStr = ep.GetRunCmd() &
@@ -1078,7 +1102,8 @@
                      "\"" + GetRemotePartyName() + "\"" &
                      PString(PString::Unsigned, duration) &
                      "\"" + codecStr + "\"" &
-                     "\"" + product + "\"";
+                     "\"" + product + "\"" &
+                     "\"" + calledParty + "\"";
     PTRACE(1, "Executing : " << cmdStr);
     system((const char *)cmdStr);
   } else {
@@ -1125,42 +1150,33 @@
                                         H323AudioCodec & codec)
 {
   codec.SetSilenceDetectionMode(H323AudioCodec::NoSilenceDetection);
-  PStringStream codecName;
-  codecName << codec;
+  PString codecName = codec.GetMediaFormat();
 
   PString ogm;
   BOOL isPCM = FALSE;
 
-  if (codec.IsDescendant(G7231_File_Codec::Class())) {
+  if (codecName == OPAL_G7231_6k3) {
     ogm   = ep.GetG7231OGM();
     isPCM = FALSE;
-  } else if (codec.IsDescendant(H323_GSM0610Codec::Class())) {
-    ogm   = ep.GetGSMOGM();
-    isPCM = TRUE;
-  } else if (codec.IsDescendant(MicrosoftGSMCodec::Class())) {
-    ogm   = ep.GetGSMOGM();
-    isPCM = TRUE;
-  } else if (codec.IsDescendant(H323_muLawCodec::Class())) {
-    ogm   = ep.GetG711OGM();
-    isPCM = TRUE;
-  } else if (codec.IsDescendant(H323_ALawCodec::Class())) {
-    ogm   = ep.GetG711OGM();
-    isPCM = TRUE;
-  } else if (codec.IsDescendant(H323_LPC10Codec::Class())) {
-    ogm   = ep.GetLPC10OGM();
-    isPCM = TRUE;
-  } else if (codec.IsDescendant(SpeexCodec::Class())) {
-    ogm   = ep.GetSPEEXOGM();
-    isPCM = TRUE;
-  } else if (codec.IsDescendant(H323_iLBC_Codec::Class())) {
-    ogm   = ep.GetiLBCOGM();
-    isPCM = TRUE;
   } else {
-    cerr << "Unknown codec \"" << codecName << endl;
-    return FALSE;
+    static OpalMediaFormat::List list = H323PluginCodecManager::GetMediaFormats();
+    if (list.GetValuesIndex(codecName) == P_MAX_INDEX) {
+      cerr << "Unknown codec \"" << codecName << endl;
+      return FALSE;
+    }
+    isPCM = TRUE;
+    if (codecName == OPAL_G711_ULAW_64K || codecName == OPAL_G711_ALAW_64K)
+      ogm = ep.GetG711OGM();
+    else if (codecName == OPAL_GSM0610)
+      ogm = ep.GetGSMOGM();
+    else if (codecName.Find("iLBC") != P_MAX_INDEX)
+      ogm = ep.GetiLBCOGM();
+    else if (codecName.Find("Speex") != P_MAX_INDEX)
+      ogm = ep.GetSPEEXOGM();
+    else
+      ogm = ep.GetG711OGM();
   }
 
-
   PWaitAndSignal mutex(connMutex);
 
   if ((recordFile == NULL) && (isEncoding == FALSE)) {
@@ -1199,8 +1215,10 @@
 
     transmitCodecName = codecName;
     if (!StartMenu(0)) {
-      if (!PFile::Exists(ogm))
+      if (!PFile::Exists(ogm)) {
         cerr << "error: cannot find OGM \"" << ogm << "\"" << endl;
+        return FALSE;
+      }
       else
         ogmChannel->QueueFile(ogm);
       if (!ep.GetNoRecordG7231())
diff -Nur openam-1.13.5/main.h openam/main.h
--- openam-1.13.5/main.h	2004-01-02 03:52:10.000000000 +0100
+++ openam/main.h	2005-02-03 02:21:42.000000000 +0100
@@ -30,6 +30,13 @@
  * Contributor(s): ______________________________________.
  *
  * $Log$
+ * Revision 1.37  2005/02/03 01:21:42  csoutheren
+ * Added called party to call log and command parameters
+ *
+ * Revision 1.36  2004/05/03 13:25:46  rjongbloed
+ * Converted everything to be codec plug in freindly
+ * Removed GSM and G.729 as now plug ins are "the way"!
+ *
  * Revision 1.35  2004/01/02 02:52:10  csoutheren
  * Thanks to Michal Zygmuntowicz for these changes
  *   Added support for iLBC codec
@@ -160,7 +167,6 @@
 #include <h323.h>
 #include <h323pdu.h>
 #include <h245.h>
-#include <gsmcodec.h>
 #include <lid.h>
 
 #include <ptclib/pwavfile.h>
@@ -326,6 +332,7 @@
     AnswerCallResponse OnAnswerCall(const PString &, const H323SignalPDU &, H323SignalPDU &);
     BOOL OnStartLogicalChannel(H323Channel & channel);
     void OnUserInputString(const PString & value);
+    BOOL OnReceivedSignalSetup(const H323SignalPDU & setupPDU);
 
     // new functions
     void StartRecording();
@@ -350,6 +357,7 @@
     PFilePath recordFn;
     PString transmitCodecName, receiveCodecName;
     BOOL    recordTrigger;
+    PString calledParty;
     PMutex  connMutex;
 
     PCM_RecordFile * recordFile;
diff -Nur openam-1.13.5/version.h openam/version.h
--- openam-1.13.5/version.h	2003-03-07 01:49:19.000000000 +0100
+++ openam/version.h	2005-03-11 01:15:52.000000000 +0100
@@ -39,7 +39,7 @@
 #define MAJOR_VERSION 1
 #define MINOR_VERSION 1
 #define BUILD_TYPE    ReleaseCode
-#define BUILD_NUMBER 18
+#define BUILD_NUMBER 19
 
 
 #endif  // _OpenAm_VERSION_H
